editor/components/document-outline/index.js

```diff
@@ -136,23 +133,20 @@ export const DocumentOutline = ( { blocks = [], title, onSelect, postType } ) =>
 };
 
 export default compose(
-	connect(
-		( state ) => {
-			return {
-				title: getEditedPostAttribute( state, 'title' ),
-				blocks: getBlocks( state ),
-				postTypeName: getCurrentPostType( state ),
-			};
-		},
-		{
-			onSelect( uid ) {
-				return selectBlock( uid );
-			},
-		},
-	),
-	withAPIData( ( { postTypeName } ) => {
+	withSelect( ( select ) => {
+		const { getEditedPostAttribute, getBlocks } = select( 'core/editor' );
+		const { getPostType } = select( 'core' );
+
+		return {
+			postType: getPostType( getEditedPostAttribute( 'type' ) ),
```

Comment:
I realized `postType` is processed only in one place so it could be also moved to the `withSelect`:
```js
isTitleSupported: get( postType, [ 'supports', 'title' ], false ),
```